From fa31b2a1d20105b5f7ad074e2807e0e9fe261823 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 15 Sep 2020 20:39:40 +0300
Subject: [PATCH] Revert "[SW-12754] gaudi: security: condition security en"

In master, we would like to have the security unconditionally, so we get
the best testing coverage we can.

This reverts commit dbab7868aa48aaea21b71e1228d4c49893490050.

Change-Id: I4a458679d4b3f2fac4d4e7ff305708eb59c7065b
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 77ac7f2..7d87a63 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -536,14 +536,6 @@
 
 	if (preboot_security_is_enabled()) {
 		preboot_puts("Security is enabled\n");
-
-		preboot_secure_cpu();
-		preboot_secure_psoc();
-		preboot_secure_pci();
-		preboot_secure_mme();
-		preboot_secure_tpc();
-
-		/* TODO: implement the rest of security enablement */
 	} else {
 		preboot_puts("Security is disabled\n");
 		/* TODO: add preboot_security_pci_disable(); */
@@ -559,6 +551,14 @@
 	preboot_puts("Disabling security temporarily till all implemented\n");
 	preboot_security_pci_disable();
 
+	preboot_secure_cpu();
+	preboot_secure_psoc();
+	preboot_secure_pci();
+	preboot_secure_mme();
+	preboot_secure_tpc();
+
+	/* TODO: implement the rest of security enablement */
+
 	cpu_boot_err_clrset(CPU_BOOT_ERR0_SECURITY_NOT_RDY, false);
 	preboot_puts("Security init finished.\n");
 }
